alerts/openshift-container-storage-operator/StorageAutoScalingFailed.md

```diff
@@ -0,0 +1,77 @@
+# StorageAutoScalingFailed
+
+## Meaning
+
+When scaling the system to accommodate additional storage, failures may occur
+due to various factors.In such instances, the `StorageAutoScalingFailed` alert
+is fired.
+
+## Impact
+
+The storage cannot be scaled at the moment.
+
+## Diagnosis
+
+There are multiple stages where the storage configurations might not have
+matched the requested storage parameters.
+
+1. OSD pod is in `Pending` state:
+    If it is in `Pending` state with the following logs, there is a high chance
+    it is due to the exhaustion of the resources.
+    ```bash
+    Warning  FailedScheduling  3m53s  default-scheduler  0/6 nodes are available: 1 Insufficient cpu, 2 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/6 nodes are available: 1 Insufficient cpu, 5 Preemption is not helpful for scheduling.
+    ```
+
+    Describe the pod in `Pending` state and check for the events.
+    ```bash
+    oc describe pod <osd_pod>
+    ```
+
+    If there are no errors, retrieve the pvc of the pod in `Pending` state.
+    ```bash
+    pvcName=$(oc get pod <osd_pod> -o jsonpath='{.spec.volumes[*].persistentVolumeClaim.claimName}')
+    oc get pvc $pvcName -o yaml
+    ```
+    Check for the health and events of the PVC for detailed information.
+
+2. PVC is in `Pending` state:
+
+    ```bash
+    oc get pvc <pvc_name> -o yaml
+    ```
+
+    Inspect the events of the pvc. Additionally, retrieve the pv of the pvc if
+    it is still in `pending` state.
+    ```bash
+    oc get pv <pvc_name> -o jsonpath='{.spec.volumeName}'
+    ```
+
+3. PV is in `Failed` state:
+    ```bash
+    oc get pv <pv_name> -o yaml
+    ```
+    If the pv is in `Failed` state, there might be a storage issue. Check the
+    events of the pv for more information.
+
+## Mitigation
+
+1. If the resources are exhausted and pods are failing, the memory of the nodes
+needs to be increased:
+    ```bash
+    oc patch storageautoscaler <resource> --type=merge --patch '{"spec":{"storageCapacityLimit":<capacity>}}'
+    ```
+
+2. Alternatively, one could try to restart the rook operator:
+    ```bash
+    oc rollout restart deployment rook-ceph-operator
+    ```
+
+    After restarting, look for errors or warnings in the logs of rook-ceph-operator:
+    ```bash
+    oc logs -l app=rook-ceph-operator
+    ```
+
+    And monitor the status of the pods and cephcluster post restart.
+
+3. Check upon the Ceph Storage: In case of issues with new OSDs, restart them
```

Comment:
also try to add in case of pv in pending state, try to look at pv errros and try to fix the 3rd party storage provider